packages/openapi-to-graphql/src/oas_3_tools.ts

```diff
@@ -906,15 +973,65 @@ export function getParameters(
 }
 
 /**
- * Returns an array of server objects for the opeartion at the given path and
+ * Returns an hash containing the callbacks defined in the given endpoint.
+ */
+export function getEndpointCallbacks(
+  path: string,
+  method: string,
+  oas: Oas3,
+  data: PreprocessingData
+): { [key: string]: CallbackObject } {
+  const callbacks = {}
+  const endpoint: OperationObject = oas.paths[path][method]
+
+  if (typeof endpoint.callbacks === 'object') {
+    let callbacksObject: CallbacksObject = endpoint.callbacks
+    for (let callbackName in callbacksObject) {
+      if (typeof callbacksObject[callbackName] === 'object') {
+        let callbackObject: CallbackObject | ReferenceObject =
+          callbacksObject[callbackName]
+        // Make sure we have CallbackObject:
+        if (typeof (callbackObject as ReferenceObject).$ref === 'string') {
+          callbackObject = resolveRef(
+            (callbackObject as ReferenceObject).$ref,
+            oas
+          ) as CallbackObject
+        } else {
+          callbackObject = (callbackObject as any) as CallbackObject
+        }
+        // console.log("CHECK CALLBACK OBJ", callbackName, callbackObject)
+
+        // Make sure CallbackObject contains PathItemObject:
+        for (let expression in callbackObject) {
+          let pathItem: PathItemObject = callbackObject[expression]
+          // console.log("CHECK CALLBACK ITEM", expression, pathItem)
+          if (typeof (pathItem as ReferenceObject).$ref === 'string') {
+            pathItem = resolveRef(callbackObject[callbackName]['$ref'], oas)
+          } else {
+            pathItem = (pathItem as any) as PathItemObject
+          }
+
+          const callback: CallbackObject = { [expression]: pathItem }
+          callbacks[callbackName] = { ...callbacks[callbackName], ...callback }
```

Comment:
As mentioned earlier, I think callbacks should have a unique key/name based on the path and the cbName, not just the cbName. I'd like to get the opinion of @ErikWittern but if we do go with this route, we should at least check if we are overwriting a callback and throw a warning if it is the case.